catalog/ui/search-ui/search-endpoint/src/test/java/org/codice/ddf/ui/searchui/query/controller/SearchControllerTest.java

```diff
@@ -258,6 +243,101 @@ public void testMetacardTypeValuesCacheEnabled() throws Exception {
         assertReplies(replies);
     }
 
+    @Test
+    public void testReducedMaxQueryResults() throws Exception {
+        final int MAX_QUERY_RESULTS = 7;
+        final int MAX_FRAMEWORK_PAGE = 17;
+
+        // the framework is set to return a single page of 17 results
+        framework = createFramework(MAX_FRAMEWORK_PAGE);
+
+        Subject subject = mock(Subject.class);
+        Search search = mock(Search.class);
+        ArgumentCaptor<QueryResponse> queryResponseArgumentCaptor = ArgumentCaptor.forClass(
+                QueryResponse.class);
+        searchController = preparePagingTestController(framework);
+
+        //this is the maximum amount of results a query should return
+        searchController.setMaxQueryResults(MAX_QUERY_RESULTS);
+        SourceQueryRunnable sourceQueryRunnable = new SourceQueryRunnable(searchController,
+                "sampleSourceId",
+                mockSearchRequest,
+                subject,
+                new HashMap<>(),
+                search,
+                mockServerSession,
+                mockFuture,
+                mockFuture);
+        sourceQueryRunnable.run();
+
+        verify(search).update(any(), queryResponseArgumentCaptor.capture());
+        assertThat("The framework should have only return " + MAX_FRAMEWORK_PAGE + " hits",
+                queryResponseArgumentCaptor.getValue()
+                        .getHits(),
+                is((long) MAX_FRAMEWORK_PAGE));
+        assertThat("The max query results was set to " + MAX_QUERY_RESULTS,
+                queryResponseArgumentCaptor.getValue()
+                        .getResults()
+                        .size(),
+                is(MAX_QUERY_RESULTS));
+    }
+
+    @Test
+    public void testPagingFromCatalog() throws Exception {
+        final int MAX_QUERY_RESULTS = 15;
+        final int MAX_FRAMEWORK_PAGE = 10;
+        final int PAGES_EXPECTED = 2;
```

Comment:
😄  I like how you extracted these values. I'd like to see more of this, as it makes it clear to someone with little context what the important inputs to the test are.